Sources/Schema.swift

```
@@ -100,7 +101,27 @@ extension SchemaBuilder {
         }
 
         let name = components[2]
+
+        if let cachedSchema = ReferenceResolver.cachedSchemaReferences[name] {
+            return Structure(name: name, structure: cachedSchema)
+        }
+        if let resolving = ReferenceResolver.resolvingReferences[name], resolving {
+            return Structure(name: name, structure: .unresolvedReference(pointer.path))
```

I would rather we throw an error here. You could make an UnresolvedReference error.

---

If it threw an error here other references in the spec wouldn't have a chance to resolve and then cache themselves, that's why this intermediate step is needed

---

Unless I'm misunderstanding, if *any* reference is unresolved, the spec should be invalid and fail to map. Could you give an example of a case where an unresolved reference should be a valid schema? What does that mean to a consumer of the library when they're trying to produce a strictly typed model?

---

You're right that it's not a valid spec, this value is just a placeholder during the first pass. It lets the resolver return _something_ if the resolving reference is currently resolving (thus stopping infinite recursion). I agree it shouldn't be part of the public API which is why I don't think this is the best implementation of this fix, but I can't see any other easy solution